Lib/uuid.py

```diff
@@ -110,6 +111,46 @@ class SafeUUID:
 _RFC_4122_VERSION_7_FLAGS = ((7 << 76) | (0x8000 << 48))
 _RFC_4122_VERSION_8_FLAGS = ((8 << 76) | (0x8000 << 48))
 
+_random_hook = None
+_time_hook = None
+
+
+def _gen_random(size):
+    if _random_hook is None:
+        return os.urandom(size)
+    else:
+        return _random_hook(size)
+
+
+def _gen_time():
+    if _time_hook is None:
+        return time.time_ns()
+    else:
+        return _time_hook()
```

Comment:
I suggest to initialize _random_hook to os.urandom and _time_hook to time.time.ns, to avoid an useless test:

```suggestion
def _gen_random(size):
    return _random_hook(size)


def _gen_time():
    return _time_hook()
```

Update also _install_py_hooks() to do the same if arguments are None.